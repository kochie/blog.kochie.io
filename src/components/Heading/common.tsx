export const Common = (
  <>
    <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    <link rel="manifest" href="/manifest.json" />
    <meta name="theme-color" content="#1f2937" />

    <link
      rel="icon"
      href="/images/icons/blog-logo-128.png"
      sizes="128x128"
      type="image/png"
    />
    <link
      rel="icon"
      href="/images/icons/blog-logo-192.png"
      sizes="192x192"
      type="image/png"
    />
    <link
      rel="icon"
      href="/images/icons/blog-logo-512.png"
      sizes="512x512"
      type="image/png"
    />
  </>
)
