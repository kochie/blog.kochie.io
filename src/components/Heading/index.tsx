import Head from 'next/head'

interface HeadingProps {
  title: string
}

export default ({ title }: HeadingProps) => {
  return (
    <Head>
      <title>{title}</title>
      <meta
        name="description"
        content="I'm Rob, this is my blog about software and ECSE engineering, and other things I find interesting!"
      />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      <link rel="manifest" href="/manifest.json" />
      <link
        rel="icon"
        href="/images/blog-logo-128.png"
        sizes="128x128"
        type="image/png"
      />
      <link
        rel="icon"
        href="/images/blog-logo-192.png"
        sizes="192x192"
        type="image/png"
      />
      <link
        rel="icon"
        href="/images/blog-logo-512.png"
        sizes="512x512"
        type="image/png"
      />
    </Head>
  )
}
