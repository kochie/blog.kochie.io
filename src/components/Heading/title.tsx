'use client'
import { useEffect } from 'react'

export default function Title({ title }: { title: string }) {
  useEffect(() => {
    document.title = title
  }, [title])

  return null
}
