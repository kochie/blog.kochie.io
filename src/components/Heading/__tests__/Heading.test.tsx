import React from 'react'
import { create } from 'react-test-renderer'

import Heading from '..'

it('renders correctly', () => {
  const tree = create(<Heading title={'testing'} />).toJSON()
  expect(tree).toMatchSnapshot()
})
