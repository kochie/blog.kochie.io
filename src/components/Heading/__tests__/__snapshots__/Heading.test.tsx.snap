// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADING COMPONENT renders correctly 1`] = `null`;
